cpp/include/cuvs/neighbors/cagra_types.hpp

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, NVIDIA CORPORATION.
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```

Comment:
Since cuVS is not going to have a header-only component, we no longer need to separate the headers out like this. This is a change that should be okay for users to make- instead of having to include 2 different files, they can jsut include `cuvs/neighbors/cagra.hpp`. And there should only be declarations in the new headers. All defintions/implementations should be compiled into the source files.